drivers: esp32: Fix esp_wifi_drv strncpy warning

rruuaanng opened the pull request:
## issue

https://github.com/zephyrproject-rtos/zephyr/issues/80066

Fixed '-Wstringop-truncation' warning thrown by ssid when using strncpy on esp32.

other code:
https://github.com/zephyrproject-rtos/zephyr/blob/6f64f77557967a41ef7df429f0a831787bb8418f/drivers/wifi/nxp/nxp_wifi_drv.c#L696-L697

This code does not do this:
https://github.com/zephyrproject-rtos/zephyr/blob/6f64f77557967a41ef7df429f0a831787bb8418f/drivers/wifi/esp32/src/esp_wifi_drv.c#L736-L740

Fixes: #80066

jukkar commented on drivers/wifi/esp32/src/esp_wifi_drv.c:
```diff
@@ -734,6 +734,7 @@ static int esp32_wifi_status(const struct device *dev, struct wifi_iface_status
 	}
 
 	strncpy(status->ssid, data->status.ssid, WIFI_SSID_MAX_LEN);
+	status->ssid[WIFI_SSID_MAX_LEN - 1] = 0;
```
This does not look right. The `status->ssid` is 32 + 1 characters long, so in order to write to the last byte, then you want to do `status->ssid[WIFI_SSID_MAX_LEN] = '\0';`

rruuaanng replied:
You're right, but when I saw that other codes were not written like this, I was a little suspicious.

rruuaanng committed:
drivers: esp32: Fix esp_wifi_drv strncpy warning

Fixed '-Wstringop-truncation' warning thrown by
ssid when using strncpy on esp32.

Signed-off-by: James Roy <rruuaanng@outlook.com>

brandon-exact commented on drivers/wifi/esp32/src/esp_wifi_drv.c:
```diff
@@ -734,6 +734,7 @@ static int esp32_wifi_status(const struct device *dev, struct wifi_iface_status
 	}
 
 	strncpy(status->ssid, data->status.ssid, WIFI_SSID_MAX_LEN);
+	status->ssid[WIFI_SSID_MAX_LEN] = '\0';
```
I am confused, AFAICT `status->ssid` is 32 bytes. currently, this change gives

```bash
/home/brandon/fork/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c: In function 'esp32_wifi_status':
/home/brandon/fork/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c:737:21: warning: array subscript 32 is above array bounds of 'char[32]' [-Warray-bounds]
  737 |         status->ssid[WIFI_SSID_MAX_LEN] = '\0';
```

rruuaanng replied:
Actually in my previous submission, it was [WIFI_SSID_MAX_LEN-1]. But it seems the auditors had issues with it, so I changed it

jukkar replied:
> @jukkar @krish2718 should we change the API from `char ssid[WIFI_SSID_MAX_LEN];` to `char ssid[WIFI_SSID_MAX_LEN + 1];` as a proper solution instead?
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/wifi_mgmt.h#L432 https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/wifi_mgmt.h#L432

Yes, that is correct. If we want to store null at the end, there needs to be space for it in the buffer (so +1 is needed).

rruuaanng replied:
I think it's good to change it to WIFI_SSID_MAX_LEN-1. Instead of changing the definition, because WIFI_SSID_MAX_LEN already indicates the maximum length. If you +1 to the maximum length, it will make the code extremely difficult to understand.

brandon-exact replied:
The standard for SSID is 32 bytes long. The way it is now, it will only be 31 bytes due to the '\0' so it must be increased if we want null termination

rruuaanng replied:
If it's okay, I'll reopen a PR tomorrow. It'll change the size of the ssid member (even though I’m not a big fan of this, but if the ssid is the standard 32-byte length, then the ssid member has to be 33 bytes).

rruuaanng commented:
I'll close this PR to apply the changes suggested by @brandon-exact.

Thank for your reviewer this :)

rruuaanng commented:
https://github.com/zephyrproject-rtos/zephyr/pull/80267
